Tasks/UseRubyVersion/userubyversion.ts

```diff
@@ -0,0 +1,101 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+
+import * as semver from 'semver';
+
+import * as task from 'vsts-task-lib/task';
+import * as tr from 'vsts-task-lib/toolrunner';
+import * as tool from 'vsts-task-tool-lib/tool';
+
+export enum Platform {
+    Windows,
+    MacOS,
+    Linux
+}
+
+export function getPlatform(): Platform {
+    switch (process.platform) {
+        case 'win32': return Platform.Windows;
+        case 'darwin': return Platform.MacOS;
+        case 'linux': return Platform.Linux;
+        default: throw Error(task.loc('PlatformNotRecognized'));
+    }
+}
+
+interface TaskParameters {
+    readonly versionSpec: string;
+    readonly outputVariable: string;
+    readonly addToPath: boolean;
+}
+
+function evaluateVersions(versions: semver.SemVer[], versionSpec: string): semver.SemVer | null {
+    task.debug('evaluating ' + versions.length + ' versions');
+    versions = versions.sort();
+    for (let i = versions.length - 1; i >= 0; i--) {
+        const version: semver.SemVer = versions[i];
+        const satisfied: boolean = semver.satisfies(version, versionSpec);
+        if (satisfied) {
+            task.debug('matched: ' + version);
+            return version;
+        }
+    }
+
+    task.debug('match not found');
+    return null;
+}
+
+export async function useRubyVersion(parameters: TaskParameters, platform: Platform): Promise<void> {
+    if (platform === Platform.Windows) {
+        const toolName: string = 'Ruby';
+        const installDir: string | null = tool.findLocalTool(toolName, parameters.versionSpec);
+        if (!installDir) {
+            // Fail and list available versions
+            throw new Error([
+                task.loc('VersionNotFound', parameters.versionSpec),
+                task.loc('ListAvailableVersions'),
+                tool.findLocalToolVersions(toolName)
+            ].join(os.EOL));
+        }
+
+        task.setVariable(parameters.outputVariable, installDir);
```

Comment:
setTaskVariable is better since if there are multiple versions of a task, the variable will get overwritten. For this task setVariable might be ok for the scenario we envision it will be used in.

Reply:
I thought the point of this was for it to be available in script tasks.  Wouldn't using a task variable here make this unavailable in those tasks?

Reply:
Good point, the way we envision this will be used, what you have is correct.